File: src/domain/entities/session/SessionRuntimeState.ts
/**
 * Durable runtime mirror of the real WhatsApp session state.
 * Unlike the in-memory host snapshot, this state is meant to survive restarts.
 */
export enum SessionRuntimeState {
  New = 'new',
  Starting = 'starting',
  Connected = 'connected',
  Reconnecting = 'reconnecting',
  Stopping = 'stopping',
  Stopped = 'stopped',
  Failed = 'failed',
  LoggedOut = 'logged_out',
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/session/SessionRuntimeState.java
package com.digows.whatsappgateway.session;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum SessionRuntimeState
{
  NEW("new"),
  STARTING("starting"),
  CONNECTED("connected"),
  RECONNECTING("reconnecting"),
  STOPPING("stopping"),
  STOPPED("stopped"),
  FAILED("failed"),
  LOGGED_OUT("logged_out");

  private final String wireValue;

  SessionRuntimeState(String wireValue)
  {
    this.wireValue = wireValue;
  }

  @JsonValue
  public String getWireValue()
  {
    return wireValue;
  }

  @JsonCreator
  public static SessionRuntimeState fromWireValue(String wireValue)
  {
    for (SessionRuntimeState value : values())
    {
      if (value.wireValue.equals(wireValue))
      {
        return value;
      }
    }

    throw new IllegalArgumentException("Unsupported session runtime state \"" + wireValue + "\".");
  }
}
